Fitness/Controllers/HomeController.swift

```diff
@@ -281,7 +286,7 @@ extension HomeController: UICollectionViewDelegate, UICollectionViewDelegateFlow
             let endDate = cal.date(bySettingHour: 23, minute: 59, second: 0, of: currDate)!
```

Comment:
I know you didn't add this, but can we change this line 
```suggestion
            let endDate = cal.date(bySettingHour: 23, minute: 59, second: 0, of: currDate) ?? Date()
```